### ISSUE TITLE
Multi Step Form Only -Save button throws 500 Error AIO Docker Image

### ISSUE THREAD

**Issue by thering1975TG** [Budibase/budibase#12951]
**Checklist**
- [x] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**

- Self
  - Method: Docker RUN AIO on Debian 12 VPS
  - Nginx Proxy Manager Docker handes all routing to budibase
  - Budibase Version: 2.17.3 and .1 version

`{
  "browser": {
    "language": "en-GB",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36 Edg/121.0.0.0",
    "platform": "Win32",
    "vendor": "Google Inc."
  },
  "server": {
    "budibaseVersion": "2.17.1",
    "hosting": "docker",
    "nodeVersion": "v20.10.0",
    "platform": "linux",
    "cpuArch": "x64",
    "cpuCores": 2,
    "cpuInfo": "QEMU Virtual CPU version 2.5+",
    "totalMemory": "1.9208831787109375GB",
    "uptime": "1 day(s), 1 hour(s), 43 minute(s)"
  }
}`

**Describe the bug**
I have two seperate forms one which is a single step single page form linked to one internal DB Table. When i test this the form saves correctly and all is good

The bigger form which is in a separate app however will not save at all and in dev console gives a 500 error (different internal DB to the single page form)

The form itself is a multi step form, with next and previous buttons. The submit button only appears on the last page. 

The next buttons are set up as follows:

![next button](https://github.com/Budibase/budibase/assets/87522339/5516e380-34e9-47a6-9a7b-e84c9e6aa73e)

- Validate was added at each stage as the Forename Initial text field is set to max 1 character without validate if more characters are added it still allows you to proceed to next step which is wrong

The Submit button is set as follows:

**Validate first**

![validate](https://github.com/Budibase/budibase/assets/87522339/4739d9df-6dc3-4a9a-8627-77cc9822615b)

**Save Next to internal DB which has pubic read write access currently for testing**

![save](https://github.com/Budibase/budibase/assets/87522339/b0fb4fbc-ac04-4ab2-a7f4-4f1d8ad250b7)

**Navigate to**

![navigate](https://github.com/Budibase/budibase/assets/87522339/8d87d038-cfd2-43bf-b37f-5af51e3c9569)

I tested also with a confirmation request on save, and that loads up the pop up correctly but then throws the same internal error when actually trying to save

The form was designed as per the Budibase You Tube video with field groups and multi step setup

**Form Layout looks like this (had to collapse the form elements as too long to screenshot**

![Layout](https://github.com/Budibase/budibase/assets/87522339/0d77d781-16f8-40f4-bf49-0b66fa509902)

I can rule out Nginx Proxy manager as the single page forms saves fine it just happens on this bigger multipsetp form and have tried to suss out whats wrong but nothing obvious stands out

Here is the Dev Console Error:

![devconsole](https://github.com/Budibase/budibase/assets/87522339/b32614a1-e151-41a4-9526-ba764cd4784f)

And if select the 2nd Critical Error i get this:

`{
    "message": "Unable to retrieve app DB - no app ID.",
    "status": 500
}`

Clicking into this it pin points the following not sure if it relates or any use

`try {
                p = await fetch(i, {
                    method: n,
                    headers: d,
                    body: h,
                    credentials: "same-origin"
                })`
And This

` console.warn(`[Client] HTTP ${t} on ${r}:${n}\n\t${a}`)
            } else
                console.error("Unhandled error from API client", e)
        }`

The 404 error appears to relate to your favicon not accesible

**Expected behavior**
Form saves and row is added to internal DB
  

**Laptop and Mobile:**
 - Windows and Android
 - Edge
 - Latest

**Comment by thering1975TG** [comment]
Additional note following some testing.

I think i may have found the issue but need to test on the original form

I duplicated the app to test and changed how it was set up:

Original app was set with the REQUIRED button on for certain fields in the internal database section. A lot of these fields are hidden and only shown when a user select a certain option in one of the questions. If the Item is shown then it is required, this is how it works on other systems i have used previously

New Version

I turned the REQUIRED option in the database column sections to OFF, to test if this was causing the issue. As soon as i done this the data saved correctly.

I am just testing the setting REQUIRED option on in the design section form elements, if it saves then i found my problem with the original setup.

Will update accordingly
